Prepare steps and examples for wallet API tests

# Problem
Prepare configuration and steps to setup status-go as daemon with wallet API exposed via RPC to facilitate testing.

# Acceptance Criteria
- Exact steps to execute to have status-go daemon running with wallet API exposed
- Order of wallet initialisation 
- Examples of wallet JSON-RPC requests

# Steps
// Run statusd with config and seed-phrase for account creation:
`./build/bin/statusd -c ./config.json -log DEBUG --seed-phrase="test test test test test test test test test test test junk"`

// config.json:
// HttpHost and HttpPort are http server address for RPC requests
```
{
    "DataDir": "/tmp/status-go-data",
    "NodeKey": "b5c65bcfdb02261052b71cb5f6dec095d0e10df649b4a2b7d881e55e306cbebc",
    "Rendezvous": false,
    "NoDiscovery": false,
    "ListenAddr": "0.0.0.0:30303",
    "HTTPEnabled": true,
    "HTTPHost": "0.0.0.0",
    "HTTPPort": 3333,
    "APIModules": "eth,admin,wallet",
    "WalletConfig": {
      "Enabled": true
    },
    "WakuConfig": {
        "Enabled": false
    },
    "Networks": [
        {
            "ChainID": 11155111,
            "RPCURL": "http://localhost:8545"
        }
    ]
}
```

// Initializes transfer controller and starts fetching transfer history
```
{
  "jsonrpc": "2.0",
  "id": 37,
  "method": "wallet_checkRecentHistoryForChainIDs",
  "params": [
    [
      11155111,
      11155420,
      421614
    ],
    [
      "0xad05399d2864f4c1ac5e51c73f022021be8c795f",
      "0xb814b99b59f319405691fd131f7db97d34f4c4a7"
    ]
  ]
}
```


// Send ETH transfer
```
{
  "jsonrpc": "2.0",
  "id": 116,
  "method": "wallet_createMultiTransaction",
  "params": [
    {
      "fromAddress": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
      "fromAmount": "0x5af3107a4000",
      "fromAsset": "ETH",
      "multiTxType": "MultiTransactionSend",
      "toAddress": "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
      "toAsset": "ETH"
    },
    [
      {
        "bridgeName": "Transfer",
        "chainID": 11155111,
        "transferTx": {
          "data": "",
          "from": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
          "gas": "0x5BBF",
          "input": "",
          "maxFeePerGas": "0xbcc0f04fd",
          "maxPriorityFeePerGas": "0x3b9aca00",
          "to": "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
          "type": "0x02",
          "value": "0x5af3107a4000"
        }
      }
    ],
    "password"
  ]
}'
```

// Until balance contract is deployed to Foundry/Hardhat/Ganache network, transactions, new transfers detection does not work
// To confirm the transaction we could check tx details (params is tx hash)
```
curl -XPOST http://127.0.0.1:3333 -H 'Content-type: application/json' -d '
{
  "jsonrpc": "2.0",
  "id": 45,
  "method": "wallet_getTxDetails",
  "params": [
      "0x1e649f26c6e76ed034f4fed001e601ee86212e571400eb0bff2417ba3c9a5506"
  ]
}
'
```

// If balance contract is deployed, we can check the multitransaction entry exists:
// params: multi transaction id returned by `createMultiTransaction` method
// NOTE: currently error is not returned from this method if no entry. Instead a dummy entry with empty id is returned
```
curl -XPOST http://127.0.0.1:3333 -H 'Content-type: application/json' -d '
{
  "jsonrpc": "2.0",
  "id": 45,
  "method": "wallet_getMultiTxDetails",
  "params": [
      1717255465976
  ]
}
'
```


// We can check pending tx details. This method is deprecated, but since new transfers are not detected, pending tx never disappears
// from pending_transactions DB table, but its status is set to "Success". Then only way to retrieve it is this method:
```
curl -XPOST http://127.0.0.1:3333 -H 'Content-type: application/json' -d '
{
  "jsonrpc": "2.0",
  "id": 45,
  "method": "wallet_getPendingTransactionsForIdentities",
  "params": [
    [
      {
        "chainId": 11155111,
        "hash": "0x1e649f26c6e76ed034f4fed001e601ee86212e571400eb0bff2417ba3c9a5506"
      }
    ]
  ]
}
'
```

// If balance contract is deployed to testing network, balances can be checked with
```
curl -XPOST http://127.0.0.1:3333 -H 'Content-type: application/json' -d '
{
  "jsonrpc": "2.0",
  "id": 195,
  "method": "wallet_fetchOrGetCachedWalletBalances",
  "params": [
    [
      "0xad05399d2864f4c1ac5e51c73f022021be8c795f"
    ]
  ]
}
'
```

# Commands dumped from Desktop app (dups removed)
[wallet_rpc_requests.txt](https://github.com/user-attachments/files/15526630/wallet_rpc_requests_send_transactions_pretty_2.txt)

I've checked out https://github.com/status-im/status-go/pull/5274, ran `make statusgo` and then `./build/bin/statusd -c ./config.json -log DEBUG --seed-phrase="test test test test test test test test test test test junk"` using config.json from the example, 

got an error ` password is required when seed phrase is provided package=status-go/cmd/statusd`

after providing password, got a crash:
```
BUG[06-03|00:02:09.504|github.com/status-im/status-go/services/wallet/collectibles/commands.go:255]       start loadOwnedCollectiblesCommand       chain=31337 account=0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
DEBUG[06-03|00:02:09.504|github.com/status-im/status-go/services/wallet/collectibles/commands.go:279]       start loadOwnedCollectiblesCommand       chain=31337 account=0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 page=0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x100dd75b8]

goroutine 905 [running]:
github.com/status-im/status-go/services/wallet/connection.(*Status).GetState(0x3733333133000000?)
	/Users/anton/Status/status-go/services/wallet/connection/status.go:55 +0x28
github.com/status-im/status-go/services/wallet/connection.(*Status).SetIsConnected(0x102872700?, 0x0)
	/Users/anton/Status/status-go/services/wallet/connection/status.go:34 +0x5c
github.com/status-im/status-go/services/wallet/collectibles.(*Manager).checkConnectionStatus(0x14000e12780, 0x14000b40140?)
	/Users/anton/Status/status-go/services/wallet/collectibles/manager.go:910 +0x11c
github.com/status-im/status-go/services/wallet/collectibles.(*Manager).FetchAllAssetsByOwner(0x14000e12780, {0x102c352d8?, 0x14000b8ad20?}, 0x7a69, {0xf3, 0x9f, 0xd6, 0xe5, 0x1a, 0xad, ...}, ...)
	/Users/anton/Status/status-go/services/wallet/collectibles/manager.go:259 +0x56c
github.com/status-im/status-go/services/wallet/collectibles.(*Manager).FetchCollectibleOwnershipByOwner(0x4?, {0x102c352d8, 0x14000b8ad20}, 0x6?, {0xf3, 0x9f, 0xd6, 0xe5, 0x1a, 0xad, ...}, ...)
	/Users/anton/Status/status-go/services/wallet/collectibles/manager.go:360 +0x50
github.com/status-im/status-go/services/wallet/collectibles.(*loadOwnedCollectiblesCommand).Run(0x14000b9e9a0, {0x102c352d8, 0x14000b8ad20})
	/Users/anton/Status/status-go/services/wallet/collectibles/commands.go:281 +0x404
github.com/status-im/status-go/services/wallet/async.(*Group).Add.func1()
	/Users/anton/Status/status-go/services/wallet/async/async.go:107 +0x3c
created by github.com/status-im/status-go/services/wallet/async.(*Group).Add in goroutine 1058
	/Users/anton/Status/status-go/services/wallet/async/async.go:106 +0x84
```

--- a/api/backend_test.go
+++ b/api/backend_test.go
@@ -1622,3 +1622,31 @@ func TestTestnetEnabledSettingOnCreateAccount(t *testing.T) {
 
 	require.NoError(t, b.Logout())
 }
+
+func TestRestoreAccountAndLogin(t *testing.T) {
+	utils.Init()
+	tmpdir := t.TempDir()
+
+	backend := NewGethStatusBackend()
+
+	// Test case 1: Valid restore account request
+	restoreRequest := &requests.RestoreAccount{
+		Mnemonic:    "test test test test test test test test test test test test",
+		FetchBackup: false,
+		CreateAccount: requests.CreateAccount{
+			DisplayName:           "Account1",
+			DeviceName:            "StatusIM",
+			Password:              "password",
+			CustomizationColor:    "0x000000",
+			BackupDisabledDataDir: tmpdir,
+		},
+	}
+	account, err := backend.RestoreAccountAndLogin(restoreRequest)
+	require.NoError(t, err)
+	require.NotNil(t, account)
+
+	// Test case 2: Invalid restore account request
+	invalidRequest := &requests.RestoreAccount{}
+	_, err = backend.RestoreAccountAndLogin(invalidRequest)
+	require.Error(t, err)
+}

--- a/api/defaults.go
+++ b/api/defaults.go
@@ -237,7 +237,11 @@ func defaultNodeConfig(installationID string, request *requests.CreateAccount, o
 		nodeConfig.LogEnabled = false
 	}
 
-	nodeConfig.Networks = BuildDefaultNetworks(&request.WalletSecretsConfig)
+	if request.TestOverrideNetworks != nil {
+		nodeConfig.Networks = request.TestOverrideNetworks
+	} else {
+		nodeConfig.Networks = BuildDefaultNetworks(&request.WalletSecretsConfig)
+	}
 
 	if request.NetworkID != nil {
 		nodeConfig.NetworkID = *request.NetworkID

--- a/cmd/statusd/main.go
+++ b/cmd/statusd/main.go
@@ -34,6 +34,7 @@ import (
 	"github.com/status-im/status-go/profiling"
 	"github.com/status-im/status-go/protocol"
 	"github.com/status-im/status-go/protocol/pushnotificationserver"
+	"github.com/status-im/status-go/protocol/requests"
 	"github.com/status-im/status-go/walletdatabase"
 )
 
@@ -86,6 +87,8 @@ var (
 	// don't change the name of this flag, https://github.com/ethereum/go-ethereum/blob/master/metrics/metrics.go#L41
 	metricsEnabled = flag.Bool("metrics", false, "Expose ethereum metrics with debug_metrics jsonrpc call")
 	metricsPort    = flag.Int("metrics-port", 9305, "Port for the Prometheus /metrics endpoint")
+	seedPhrase     = flag.String("seed-phrase", "", "Seed phrase for account creation")
+	password       = flag.String("password", "", "Password for account")
 )
 
 // All general log messages in this package should be routed through this logger.
@@ -107,6 +110,12 @@ func main() {
 	if flag.NArg() > 0 {
 		printUsage()
 		logger.Error("Extra args in command line: %v", flag.Args())
+		os.Exit(1)
+	}
+
+	if *seedPhrase != "" && *password == "" {
+		printUsage()
+		logger.Error("password is required when seed phrase is provided")
 		os.Exit(1)
 	}
 
@@ -162,12 +171,6 @@ func main() {
 	}
 
 	backend := api.NewGethStatusBackend()
-	err = backend.AccountManager().InitKeystore(config.KeyStoreDir)
-	if err != nil {
-		logger.Error("Failed to init keystore", "error", err)
-		return
-	}
-
 	if config.NodeKey == "" {
 		logger.Error("node key needs to be set if running a push notification server")
 		return
@@ -186,96 +189,119 @@ func main() {
 		return
 	}
 
-	err = createDirsFromConfig(config)
-	if err != nil {
-		logger.Error("failed to create directories", "error", err)
-		return
-	}
+	if *seedPhrase != "" {
+		// Remove data inside dir to avoid conflicts with existing data or account restoration fails
+		if err := os.RemoveAll(config.DataDir); err != nil {
+			logger.Error("failed to remove data dir", "error", err)
+			return
+		}
 
-	appDB, walletDB, err := openDatabases(config.DataDir + "/" + installationID.String())
-	if err != nil {
-		log.Error("failed to open databases")
-		return
-	}
+		if err := createDirsFromConfig(config); err != nil {
+			logger.Error("failed to create directories", "error", err)
+			return
+		}
 
-	backend.StatusNode().SetAppDB(appDB)
-	backend.StatusNode().SetWalletDB(walletDB)
+		request := requests.RestoreAccount{
+			Mnemonic:    *seedPhrase,
+			FetchBackup: false,
+			CreateAccount: requests.CreateAccount{
+				DisplayName:           "Account1",
+				DeviceName:            "StatusIM",
+				Password:              *password,
+				CustomizationColor:    "0x000000",
+				BackupDisabledDataDir: config.DataDir,
+				APIConfig: &requests.APIConfig{
+					HTTPHost:   config.HTTPHost,
+					HTTPPort:   config.HTTPPort,
+					APIModules: config.APIModules,
+				},
+				NetworkID:            &config.NetworkID,
+				TestOverrideNetworks: config.Networks,
+			},
+		}
 
-	err = backend.StartNode(config)
-	if err != nil {
-		logger.Error("Node start failed", "error", err)
-		return
-	}
-
-	err = sdnotify.Ready()
-	if err == sdnotify.ErrSdNotifyNoSocket {
-		logger.Debug("sd_notify socket not available")
-	} else if err != nil {
-		logger.Warn("sd_notify READY call failed", "error", err)
+		_, err := backend.RestoreAccountAndLogin(&request)
+		if err != nil {
+			logger.Error("failed to import account", "error", err)
+			return
+		}
 	} else {
-		// systemd aliveness notifications, affects only Linux
-		go startSystemDWatchdog()
-	}
-
-	// handle interrupt signals
-	interruptCh := haltOnInterruptSignal(backend.StatusNode())
-
-	// Start collecting metrics. Metrics can be enabled by providing `-metrics` flag
-	// or setting `gethmetrics.Enabled` to true during compilation time:
-	// https://github.com/status-im/go-ethereum/pull/76.
-	if *metricsEnabled || gethmetrics.Enabled {
-		go startCollectingNodeMetrics(interruptCh, backend.StatusNode())
-		go gethmetrics.CollectProcessMetrics(3 * time.Second)
-		go metrics.NewMetricsServer(*metricsPort, gethmetrics.DefaultRegistry).Listen()
-	}
-
-	// Check if profiling shall be enabled.
-	if *pprofEnabled {
-		profiling.NewProfiler(*pprofPort).Go()
-	}
-
-	if config.PushNotificationServerConfig.Enabled {
-		options := []protocol.Option{
-			protocol.WithPushNotifications(),
-			protocol.WithPushNotificationServerConfig(&pushnotificationserver.Config{
-				Enabled:   config.PushNotificationServerConfig.Enabled,
-				Identity:  config.PushNotificationServerConfig.Identity,
-				GorushURL: config.PushNotificationServerConfig.GorushURL,
-			}),
-			protocol.WithDatabase(appDB),
-			protocol.WithWalletDatabase(walletDB),
-			protocol.WithTorrentConfig(&config.TorrentConfig),
-			protocol.WithWalletConfig(&config.WalletConfig),
-			protocol.WithAccountManager(backend.AccountManager()),
-		}
-
-		messenger, err := protocol.NewMessenger(
-			config.Name,
-			identity,
-			gethbridge.NewNodeBridge(backend.StatusNode().GethNode(), backend.StatusNode().WakuService(), backend.StatusNode().WakuV2Service()),
-			installationID.String(),
-			nil,
-			options...,
-		)
+		appDB, walletDB, err := startNode(config, backend, installationID)
 		if err != nil {
-			logger.Error("failed to create messenger", "error", err)
+			logger.Error("failed to start node", "error", err)
 			return
 		}
 
-		err = messenger.Init()
-		if err != nil {
-			logger.Error("failed to init messenger", "error", err)
-			return
+		err = sdnotify.Ready()
+		if err == sdnotify.ErrSdNotifyNoSocket {
+			logger.Debug("sd_notify socket not available")
+		} else if err != nil {
+			logger.Warn("sd_notify READY call failed", "error", err)
+		} else {
+			// systemd aliveness notifications, affects only Linux
+			go startSystemDWatchdog()
 		}
 
-		// This will start the push notification server as well as
-		// the config is set to Enabled
-		_, err = messenger.Start()
-		if err != nil {
-			logger.Error("failed to start messenger", "error", err)
-			return
+		// handle interrupt signals
+		interruptCh := haltOnInterruptSignal(backend.StatusNode())
+
+		// Start collecting metrics. Metrics can be enabled by providing `-metrics` flag
+		// or setting `gethmetrics.Enabled` to true during compilation time:
+		// https://github.com/status-im/go-ethereum/pull/76.
+		if *metricsEnabled || gethmetrics.Enabled {
+			go startCollectingNodeMetrics(interruptCh, backend.StatusNode())
+			go gethmetrics.CollectProcessMetrics(3 * time.Second)
+			go metrics.NewMetricsServer(*metricsPort, gethmetrics.DefaultRegistry).Listen()
 		}
-		go retrieveMessagesLoop(messenger, 300*time.Millisecond, interruptCh)
+
+		// Check if profiling shall be enabled.
+		if *pprofEnabled {
+			profiling.NewProfiler(*pprofPort).Go()
+		}
+
+		if config.PushNotificationServerConfig.Enabled {
+			options := []protocol.Option{
+				protocol.WithPushNotifications(),
+				protocol.WithPushNotificationServerConfig(&pushnotificationserver.Config{
+					Enabled:   config.PushNotificationServerConfig.Enabled,
+					Identity:  config.PushNotificationServerConfig.Identity,
+					GorushURL: config.PushNotificationServerConfig.GorushURL,
+				}),
+				protocol.WithDatabase(appDB),
+				protocol.WithWalletDatabase(walletDB),
+				protocol.WithTorrentConfig(&config.TorrentConfig),
+				protocol.WithWalletConfig(&config.WalletConfig),
+				protocol.WithAccountManager(backend.AccountManager()),
+			}
+
+			messenger, err := protocol.NewMessenger(
+				config.Name,
+				identity,
+				gethbridge.NewNodeBridge(backend.StatusNode().GethNode(), backend.StatusNode().WakuService(), backend.StatusNode().WakuV2Service()),
+				installationID.String(),
+				nil,
+				options...,
+			)
+			if err != nil {
+				logger.Error("failed to create messenger", "error", err)
+				return
+			}
+
+			err = messenger.Init()
+			if err != nil {
+				logger.Error("failed to init messenger", "error", err)
+				return
+			}
+
+			// This will start the push notification server as well as
+			// the config is set to Enabled
+			_, err = messenger.Start()
+			if err != nil {
+				logger.Error("failed to start messenger", "error", err)
+				return
+			}
+			go retrieveMessagesLoop(messenger, 300*time.Millisecond, interruptCh)
+		}
 	}
 
 	gethNode := backend.StatusNode().GethNode()
@@ -402,6 +428,7 @@ Examples:
   statusd -c ./default.json                      # run node with configuration specified in ./default.json file
   statusd -c ./default.json -c ./standalone.json # run node with configuration specified in ./default.json file, after merging ./standalone.json file
   statusd -c ./default.json -metrics             # run node with configuration specified in ./default.json file, and expose ethereum metrics with debug_metrics jsonrpc call
+  statusd -c ./default.json -log DEBUG --seed-phrase="test test test test test test test test test test test junk" --password=password # run node with configuration specified in ./default.json file, and import account with seed phrase and password
 
 Options:
 `
@@ -482,4 +509,35 @@ func createDirsFromConfig(config *params.NodeConfig) error {
 	}
 
 	return nil
+}
+
+func startNode(config *params.NodeConfig, backend *api.GethStatusBackend, installationID uuid.UUID) (*sql.DB, *sql.DB, error) {
+	err := backend.AccountManager().InitKeystore(config.KeyStoreDir)
+	if err != nil {
+		logger.Error("Failed to init keystore", "error", err)
+		return nil, nil, err
+	}
+
+	err = createDirsFromConfig(config)
+	if err != nil {
+		logger.Error("failed to create directories", "error", err)
+		return nil, nil, err
+	}
+
+	appDB, walletDB, err := openDatabases(config.DataDir + "/" + installationID.String())
+	if err != nil {
+		log.Error("failed to open databases")
+		return nil, nil, err
+	}
+
+	backend.StatusNode().SetAppDB(appDB)
+	backend.StatusNode().SetWalletDB(walletDB)
+
+	err = backend.StartNode(config)
+	if err != nil {
+		logger.Error("Node start failed", "error", err)
+		return nil, nil, err
+	}
+
+	return appDB, walletDB, nil
 }

--- a/protocol/requests/create_account.go
+++ b/protocol/requests/create_account.go
@@ -4,6 +4,7 @@ import (
 	"github.com/pkg/errors"
 
 	utils "github.com/status-im/status-go/common"
+	"github.com/status-im/status-go/params"
 )
 
 var ErrCreateAccountInvalidDisplayName = errors.New("create-account: invalid display name")
@@ -56,8 +57,9 @@ type CreateAccount struct {
 
 	// Deprecated: CurrentNetwork is deprecated. It was passed and not used, so nothing should be passed instead.
 	// If you want to use non-default network, use NetworkID.
-	CurrentNetwork string  `json:"currentNetwork"`
-	NetworkID      *uint64 `json:"networkId"`
+	CurrentNetwork       string           `json:"currentNetwork"`
+	NetworkID            *uint64          `json:"networkId"`
+	TestOverrideNetworks []params.Network `json:"-"` // This is used for testing purposes only
 
 	TestNetworksEnabled bool `json:"testNetworksEnabled"`
 

--- a/services/wallet/history/service.go
+++ b/services/wallet/history/service.go
@@ -486,11 +486,15 @@ func (s *Service) startTransfersWatcher() {
 			return
 		}
 
+		network := s.networkManager.Find(chainID)
+		if network == nil {
+			log.Error("Network not found", "chainID", chainID)
+			return
+		}
+
 		transferDB := transfer.NewDB(s.db)
 
 		for _, address := range addresses {
-			network := s.networkManager.Find(chainID)
-
 			transfers, err := transferDB.GetTransfersByAddressAndBlock(chainID, address, block, 1500) // 1500 is quite arbitrary and far from real, but should be enough to cover all transfers in a block
 			if err != nil {
 				log.Error("Error getting transfers", "chainID", chainID, "address", address.String(), "err", err)

